### ISSUE TITLE
How to apply grumphp for specific current commit files

### ISSUE THREAD

**Issue by sunilit42** [phpro/grumphp#448]
Hello,

we are using grumphp and it is very good.

but we have facing some issue.

It is checking all files, only only current commit files.

Looking  forward some solution.

**Comment by veewee** [comment]
Hi @sunilit42,

Sadly I don't fully understand your question. Can you rephrase?

**Comment by Landerstraeten** [comment]
Hi @sunilit42,

Do you mean checking **staged** files?

**Comment by sunilit42** [comment]
yes

**Comment by veewee** [comment]
The `run` command checks all files in git.
The `git:pre-commit` command checks staged files.

**Comment by jigneshthummar** [comment]
@veewee  thanks grumphp run command helped..

is it possible to specify specific directory to check?? we basically wants to check it in our custom modules, and wants to exclude 3rd party modules..

**Comment by jigneshthummar** [comment]
Also we are using following ignore_patterns but it seems like its not taking effect like other rules where same pattern (uncommented one) works

## The Phpcs task will sniff your code for bad coding standards.            
        phpcs:
            standard: "~/.composer/vendor/magento-ecg/coding-standard/EcgM2/ruleset.xml"
            #show_warnings: true
            tab_width: 4
            whitelist_patterns: ['.']
            encoding: ~
            #ignore_patterns: ["*/dev/*","*/vendor/*","*/generated/*","*/var/*","*/lib/*","*/update/*","*/bin/*","*/pub/*","*/setup/*","*/app/code/*"]
            ignore_patterns: ['dev','vendor','generated','var','lib','update','bin','pub','setup','app/code']
            sniffs: []
            triggered_by: [php,phtml]
            metadata:
                #priority: 870
